gapis/gfxapi/gfxapi.proto

```diff
@@ -102,6 +92,26 @@ enum UniformType {
 	Double = 4;
 }
 
+// ResourceData represents the resource state at a single point in a capture
+message ResourceData {
+	oneof data {
+		Texture texture = 1;
+		Shader shader = 2;
+		Program program = 3;
+	}
+}
+
+// Texture represents a texture resource.
+message Texture {
+	oneof type {
```

Comment:
I would vote for separate structures, so it's easier to extend them. For example, it would be helpful to communicate to the client whether a cube texture is considered complete by GL. Having separate structures keeps things like that cleaner.